src/routes/volume/detail/index.js

```diff
@@ -462,6 +468,32 @@ function VolumeDetail({ snapshotModal, dispatch, backup, engineimage, eventlog,
     },
   }
 
+  const confirmModalWithWorkloadProps = {
+    visible: confirmModalWithWorkloadVisible,
+    title: `Detach volume ${selectedVolume.name}`,
+    onOk() {
+      if (selectedVolume.actions && selectedVolume.actions.detach) {
+        dispatch({
+          type: 'volume/detach',
+          payload: {
+            url: selectedVolume.actions.detach,
+          },
+        })
+      }
+      dispatch({
+        type: 'volume/hideConfirmDetachWithWorkload',
+      })
+      dispatch({
+        type: 'volume/stopPolling',
```

Comment:
Do we have this callback? I did not see it in models/volume.js

Reply:
Thank you for your careful inspection, I did make a mistake in this position. After we detach the volume in the details page, we need to stop requesting snapshotdata. Otherwise, the back-end will response an error. So we need to call snapshotModal/stopPolling.

It’s okay without this callback, but it’s more rigorous if there is one.